spec/models/immunisation_import_row_spec.rb

```diff
@@ -1352,6 +1352,69 @@
 
         it { should eq("A Clinic") }
       end
+
+      context "with an existing community clinic" do
+        let(:data) do
+          valid_data.merge(
+            "CLINIC_NAME" => "A clinic",
+            "CARE_SETTING" => "2",
+            "SESSION_ID" => session.id.to_s
+          )
+        end
+
+        let(:session) { create(:session, organisation:, programmes:) }
+
+        before { create(:community_clinic, name: "A clinic", organisation:) }
+
+        it "is matching" do
+          immunisation_import_row.valid?
+          expect(immunisation_import_row.errors.full_messages).not_to include(
+            a_string_including("Enter a clinic name")
+          )
+        end
+      end
+
+      context "with incorrect casing for an existing clinic" do
+        let(:data) do
+          valid_data.merge(
+            "CLINIC_NAME" => "a cLinIC",
+            "CARE_SETTING" => "2",
+            "SESSION_ID" => session.id.to_s
+          )
+        end
+
+        let(:session) { create(:session, organisation:, programmes:) }
+
+        before { create(:community_clinic, name: "A clinic", organisation:) }
+
+        it "is case insensitive" do
+          immunisation_import_row.valid?
+          expect(immunisation_import_row.errors.full_messages).not_to include(
+            a_string_including("Enter a clinic name")
+          )
+        end
+      end
+
+      context "with a non-existent clinic" do
+        let(:data) do
+          valid_data.merge(
+            "CLINIC_NAME" => "A wrong clinic",
+            "CARE_SETTING" => "2",
+            "SESSION_ID" => session.id.to_s
+          )
+        end
+
+        let(:session) { create(:session, organisation:, programmes:) }
+
+        before { create(:community_clinic, name: "A clinic", organisation:) }
+
+        it "has errors" do
+          expect(immunisation_import_row).to be_invalid
+          expect(immunisation_import_row.errors.full_messages).to include(
+            a_string_including("Enter a clinic name")
+          )
```

Comment:
To be sure the error is attached to the right field, I would suggest something like:

```suggestion
          expect(immunisation_import_row.errors["CLINIC_NAME"]).to contain_exactly("Enter a clinic name")
```